packages/tools/pluggable-widgets-tools/configs/rollup.config.js

```diff
@@ -251,7 +257,7 @@ export default async args => {
                 async () => {
                     mkdirSync(mpkDir, { recursive: true });
                     await zip(outDir, mpkFile);
-                    if (!production && projectPath) {
+                    if (projectPath) {
```

Comment:
The intention was never copy release build, because they are for marketplace.. so i am not sure if this should be removed

Reply:
@tamelbrun do you have any concerns about?

Reply:
I'd like to add here that this step (so zip + cp + cp) forms the majority (90%+) of the build time. Can we be smarter about this? Could we, for example, output to the deployment path directly? Can we write out the zip into the widgetsPath directly, avoiding a cp?

Reply:
We saw this with widgets from Hedwig and Marcel, which had big node_modules folder, and the build time was multiple minutes

Reply:
Ah, good one, Kevin!

Regarding the issue about copying to a `projectPath` for production builds, could something like the following help avoid unexpected production copying:

`if ((!production && projectPath) || (production && MX_PROJECT_PATH)) {`

It would be really helpful to have the ability to automatically copying release-built .`mpk`s to a dir - for testing purposes

Reply:
Why are we testing with production build exactly?

Reply:
Npm start for example which is the one constantly copying files is just dev mode only..

Reply:
`Why are we testing with production build exactly?` Testing should be done with an artefact that's as close to the end product as possible.
In this case, tests were done against the dev build (which worked fine), but the release build caused issues.

Reply:
@KevinVlaanderen We can create the zip in the correct location immediately, but unless the zip is really huge, copying a single file is probably not the bottleneck. And I don't think we can output to the deployment folder directly, because then we have no clean folder to create the zip from. Or am I missing something here?

Reply:
To come back to the initial question: copying the widget output to the deployment folder in production/release mode seems like an unnecessary and potentially slow step that we would like to avoid. (Copying the mpk is probably not a bottleneck here either, but it also seems unnecessary.)

Reply:
If it's a 400MB file, it is a problem. Obviously that shouldn't happen, but we've seen 2 cases now where it did.
I do know for a fact that a combination of these 3 actions is responsible for the slowness in the build; all the rest (transpilation, minifying, extract native deps) is fast.

Reply:
If we have a concrete use case for outputting the mpk file to a different location, I would add an environment variable (similar to the existing MPKOUTPUT) to override the default location. This wouldn't affect any of the existing logic and allows us to keep `if (!production && projectPath) {` as it is.

Reply:
In non-production mode, creating the mpk in the widgets folder of the project (if so configured) instead of the dist folder (which would then only be the default mpk output location in production mode) would be a reasonable optimization. But I think that might be all we can squeeze out of it :).

Reply:
Thanks for the input guys! I've reverted the initial change and optimised the build step.

- `npm run watch` builds directly into deployment folder
- `npm run build` builds in dist/ and copies to both deployment and widget folder
- `npm run release` builds in dist/

After discussing with Diego we decided to add an additional command to the repo root where we copy over the widgets, so we can keep it separated from pluggable-widgets-tools

Reply:
On second thought, we might want the `watch` command to also zip into the projects' `widget/` folder? Otherwise running `watch` and rerunning || stop/start modeller will remove your changes. What do you guys think?

Reply:
The problem is that we _cannot_ create a mpk when building directly into the deployment folder, because there is no clean folder to create mpk zip from. We do if we build to `dist` first, but then we also have the extra copy step again.

Speaking of which, why would `npm run build` copy to the deployment folder? Copying the mpk to the widget folder should be enough, right?

Reply:
You're right @tamelbrun. 
I've reverted my changes a bit as we concluded that we do need `watch` to add the widget to the `widgets/` folder too. And because some people have their project on a different drive (or Dropbox :exploding_head: ) it's probably quicker to still build the mpk in `dist/` and then copy the file over, instead of building directly into the project.
Pretty sure we can do this conditionally based on if the path is on a different drive / Dropbox or not. I'll keep this in mind for the next iteration, as other stories are calling me 😄 

I've also made sure `build` doesn't copy to the deployment directory anymore :)